src/Temporalio/Bridge/OptionsExtensions.cs

```diff
@@ -450,16 +450,66 @@ public static Interop.WorkerOptions ToInteropOptions(
             {
                 throw new ArgumentException("Task queue must be provided in worker options");
             }
-            var buildId = options.BuildId;
+#pragma warning disable 0618
+            var buildId = options.DeploymentOptions?.Version?.BuildId ?? options.BuildId;
+#pragma warning restore 0618
+            var buildIdAutoDetermined = false;
             if (buildId == null)
             {
-                if (options.UseWorkerVersioning)
-                {
-                    throw new ArgumentException("BuildId must be explicitly set when UseWorkerVersioning is true");
-                }
                 var entryAssembly = Assembly.GetEntryAssembly() ??
                     throw new ArgumentException("Unable to get assembly manifest ID for build ID");
                 buildId = entryAssembly.ManifestModule.ModuleVersionId.ToString();
+                buildIdAutoDetermined = true;
+            }
+            Interop.WorkerVersioningStrategy versioningStrategy;
+            if (options.DeploymentOptions != null)
+            {
+                if (options.DeploymentOptions.Version == null)
```

Comment:
Should we also check that `buildIdAutodetermined` is `true`? We're not even using the `buildId` local var when using deployment options so why default it above? Revisiting this, I think the logic needs to change to more clearly differentiate deployment options vs non-deployment options.

Reply:
It never gets autodetermined here because they have to provide version, which means they have to provide build id. The autodetermined thing is used for the `UseWorkerVersioning`/fallback cases. 

I have added the stuff to enforce mutual exclusion which makes it more clear

Reply:
I see. I wonder if we should check that `options.BuildId` is null then if it is ignored in these cases

Reply:
Git or github did something really weird with this review. I had a bunch of changes that just didn't get published even though I have them locally and everything is pushed.

I had all this in here after my first commit with review comments but it's just magically not there. I'll figure out how to get it actually be in sync